### PR TITLE
treewide: rename name to pname&version

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Description of changes
#103997

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Stunkymonkey** [comment]
Result of `nixpkgs-review pr 165297` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>clfswm</li>
    <li>ncbi_tools</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>mucommander</li>
  </ul>
</details>
<details>
  <summary>68 packages built:</summary>
  <ul>
    <li>OVMF</li>
    <li>OVMFFull</li>
    <li>alpine-make-vm-image</li>
    <li>arc-icon-theme</li>
    <li>bfg-repo-cleaner</li>
    <li>brise</li>
    <li>bsdgames</li>
    <li>charliecloud</li>
    <li>cloud-init</li>
    <li>cloud-utils</li>
    <li>docker</li>
    <li>docker-gc</li>
    <li>docker-proxy</li>
    <li>faba-icon-theme</li>
    <li>faba-mono-icons</li>
    <li>fcitx-engines.rime</li>
    <li>fityk</li>
    <li>fn-cli</li>
    <li>fricas</li>
    <li>frostwire</li>
    <li>git-secret</li>
    <li>gnome.gnome-boxes</li>
    <li>gosmore</li>
    <li>gvisor</li>
    <li>gvisor-containerd-shim</li>
    <li>isabelle</li>
    <li>jameica</li>
    <li>lighthouse</li>
    <li>magnetophonDSP.faustCompressors</li>
    <li>miscfiles</li>
    <li>moka-icon-theme</li>
    <li>nvidia-docker</li>
    <li>out-of-tree</li>
    <li>p2pvc</li>
    <li>pdfpc</li>
    <li>pipework</li>
    <li>python310Packages.jupyter-repo2docker</li>
    <li>python39Packages.jupyter-repo2docker</li>
    <li>qemu-utils</li>
    <li>qmetro</li>
    <li>quickemu</li>
    <li>rtl-ais</li>
    <li>rxvt-unicode</li>
    <li>rxvt-unicode-emoji</li>
    <li>scilab-bin</li>
    <li>sdrplay</li>
    <li>shared_desktop_ontologies</li>
    <li>sleepyhead</li>
    <li>slmenu</li>
    <li>soapysdrplay</li>
    <li>soundmodem</li>
    <li>stdmanpages</li>
    <li>taffybar</li>
    <li>tango-icon-theme</li>
    <li>termonad</li>
    <li>tlaps</li>
    <li>tmuxPlugins.urlview</li>
    <li>tuhi</li>
    <li>urlview</li>
    <li>virt-viewer</li>
    <li>vmware-horizon-client</li>
    <li>xcircuit</li>
    <li>xfce.xfce4-icon-theme</li>
    <li>xmonad-with-packages</li>
    <li>xmr-stak</li>
    <li>xournal</li>
    <li>z3_4_4_0</li>
    <li>zscroll</li>
  </ul>
</details>

File: pkgs/applications/misc/gosmore/default.nix [hunk 1]
```diff
@@ -1,10 +1,10 @@
 { lib, stdenv, fetchsvn, libxml2, gtk2, curl, pkg-config } :
 
 let
```
Comment (davidak) [inline pkgs/applications/misc/gosmore/default.nix:3]: you can also remove these
```suggestion
```

File: pkgs/applications/misc/gosmore/default.nix [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
-  version = "31801";
 in
```
Comment (davidak) [inline pkgs/applications/misc/gosmore/default.nix:4]:
```suggestion
```

**Review by davidak — CHANGES_REQUESTED** [review]
Changes look good.

Please consider my suggestions.

Then i would build a last time and merge.

Thanks for cleaning up nixpkgs! :sparkles:

File: pkgs/applications/virtualization/virt-viewer/default.nix [hunk 3]
```diff
@@ -35,12 +35,11 @@ assert spiceSupport -> (
...
 
 stdenv.mkDerivation rec {
-  baseName = "virt-viewer";
+  pname = "virt-viewer";
   version = "11.0";
-  name = "${baseName}-${version}";
 
   src = fetchurl {
-    url = "http://virt-manager.org/download/sources/${baseName}/${name}.tar.xz";
+    url = "http://virt-manager.org/download/sources/virt-viewer/virt-viewer-${version}.tar.xz";
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/virt-viewer/default.nix:42]:
```suggestion
    url = "https://releases.pagure.org/virt-viewer/virt-viewer-${version}.tar.xz";
```

File: pkgs/applications/science/math/scilab-bin/default.nix [hunk 4]
```diff
@@ -1,11 +1,9 @@
 { stdenv, fetchurl, lib, xorg }:
 
 let
-  name = "scilab-bin-${ver}";
+  version = "6.1.1";
 
-  ver = "6.1.1";
-
-  badArch = throw "${name} requires i686-linux or x86_64-linux";
+  badArch = throw "scilab-bin-${version} requires i686-linux or x86_64-linux";
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/scilab-bin/default.nix:6]:
```suggestion
  badArch = throw "scilab-bin requires i686-linux or x86_64-linux";
```

File: pkgs/applications/science/math/scilab-bin/default.nix [hunk 5]
```diff
@@ -1,11 +1,9 @@
 { stdenv, fetchurl, lib, xorg }:
 
 let
-  name = "scilab-bin-${ver}";
+  version = "6.1.1";
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/scilab-bin/default.nix:5]:
```suggestion
```

File: pkgs/applications/science/math/scilab-bin/default.nix [hunk 6]
```diff
@@ -16,10 +14,11 @@ let
       badArch;
 in
 stdenv.mkDerivation {
-  inherit name;
+  pname = "scilab-bin";
+  inherit version;
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/scilab-bin/default.nix:18]:
```suggestion
  version = "6.1.1";
```

File: pkgs/applications/science/math/fricas/default.nix [hunk 7]
```diff
@@ -1,14 +1,11 @@
 { lib, stdenv, fetchurl, sbcl, libX11, libXpm, libICE, libSM, libXt, libXau, libXdmcp }:
 
-let
+stdenv.mkDerivation rec {
   version = "1.3.7";
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/fricas/default.nix:4]:
```suggestion
```

File: pkgs/applications/science/math/fricas/default.nix [hunk 8]
```diff
@@ -1,14 +1,11 @@
...
-  name = "fricas-" + version;
-in
-stdenv.mkDerivation {
-  inherit name;
+  pname = "fricas";
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/fricas/default.nix:5]:
```suggestion
  pname = "fricas";
  version = "1.3.7";
```

**Comment by davidak** [comment]
Result of `nixpkgs-review pr 165297` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>clfswm</li>
    <li>ncbi_tools</li>
    <li>pdfdiff</li>
  </ul>
</details>
<details>
  <summary>69 packages failed to build:</summary>
  <ul>
    <li>OVMF</li>
    <li>OVMFFull</li>
    <li>alpine-make-vm-image</li>
    <li>arc-icon-theme</li>
    <li>bfg-repo-cleaner</li>
    <li>brise</li>
    <li>bsdgames</li>
    <li>charliecloud</li>
    <li>cloud-init</li>
    <li>cloud-utils</li>
    <li>docker (docker-edge ,docker_20_10)</li>
    <li>docker-gc</li>
    <li>docker-proxy</li>
    <li>faba-icon-theme</li>
    <li>faba-mono-icons</li>
    <li>fcitx-engines.rime</li>
    <li>fityk</li>
    <li>fn-cli</li>
    <li>fricas</li>
    <li>frostwire</li>
    <li>git-secret</li>
    <li>gnome.gnome-boxes</li>
    <li>gosmore</li>
    <li>gvisor</li>
    <li>gvisor-containerd-shim</li>
    <li>isabelle</li>
    <li>jameica</li>
    <li>lighthouse</li>
    <li>magnetophonDSP.faustCompressors</li>
    <li>miscfiles</li>
    <li>moka-icon-theme</li>
    <li>mucommander</li>
    <li>nvidia-docker</li>
    <li>out-of-tree</li>
    <li>p2pvc</li>
    <li>pdfpc</li>
    <li>pipework</li>
    <li>python310Packages.jupyter-repo2docker</li>
    <li>python39Packages.jupyter-repo2docker</li>
    <li>qemu-utils</li>
    <li>qmetro</li>
    <li>quickemu</li>
    <li>rtl-ais</li>
    <li>rxvt-unicode</li>
    <li>rxvt-unicode-emoji</li>
    <li>scilab-bin</li>
    <li>sdrplay</li>
    <li>shared_desktop_ontologies</li>
    <li>sleepyhead</li>
    <li>slmenu</li>
    <li>soapysdrplay</li>
    <li>soundmodem</li>
    <li>stdmanpages</li>
    <li>taffybar</li>
    <li>tango-icon-theme</li>
    <li>termonad (termonad-with-packages)</li>
    <li>tlaps</li>
    <li>tmuxPlugins.urlview</li>
    <li>tuhi</li>
    <li>urlview</li>
    <li>virt-viewer</li>
    <li>vmware-horizon-client</li>
    <li>xcircuit</li>
    <li>xfce.xfce4icontheme (xfce.xfce4-icon-theme)</li>
    <li>xmonad-with-packages</li>
    <li>xmr-stak</li>
    <li>xournal</li>
    <li>z3_4_4_0</li>
    <li>zscroll</li>
  </ul>
</details>

**Comment by davidak** [comment]
Oh no, i get such an error again.

```
error: attribute 'xfce4icontheme' missing

       at /home/davidak/.cache/nixpkgs-review/pr-165297/build.nix:68:9:

           67|         xcircuit
           68|         xfce."xfce4icontheme"
             |         ^
           69|         xmonad-with-packages
(use '--show-trace' to show detailed location information)
```

**Comment by davidak** [comment]
A was able to build a single package, so i assume the other will also build.

`[nix-shell:~/.cache/nixpkgs-review/pr-165297]$ nix-build nixpkgs/ -A faba-icon-theme`

**Review by davidak — APPROVED** [review]
Changes look good

Should build

**Pull request merged**